### ISSUE TITLE
"Getting started" port never opens

### ISSUE THREAD

**Issue by childofsoong** [Netflix/atlas#612]
I'm trying to follow the "Getting started" instructions.  I follow the two instructions to download with curl, and then run with java:

```
$ curl -LO https://github.com/Netflix/atlas/releases/download/v1.5.2/atlas-1.5.2-standalone.jar
$ java -jar atlas-1.5.2-standalone.jar 
2017-06-05T19:05:21.355 INFO  [main] com.netflix.iep.guice.GuiceHelper: guice injector created successfully
2017-06-05T19:05:21.776 INFO  [main] com.netflix.atlas.akka.ActorService$$anonfun$startImpl$1: created actor 'akka://atlas/user/deadLetterStats' using class 'com.netflix.atlas.akka.DeadLetterStatsActor'
2017-06-05T19:05:21.788 INFO  [main] com.netflix.atlas.akka.ActorService$$anonfun$startImpl$1: created actor 'akka://atlas/user/db' using class 'com.netflix.atlas.webapi.LocalDatabaseActor'
2017-06-05T19:05:21.791 INFO  [main] com.netflix.atlas.akka.WebServer: starting WebServer on port 7101
2017-06-05T19:05:21.910 INFO  [atlas-akka.actor.default-dispatcher-9] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:21.911 INFO  [atlas-akka.actor.default-dispatcher-3] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:21.914 INFO  [atlas-akka.actor.default-dispatcher-8] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:21.916 INFO  [atlas-akka.actor.default-dispatcher-9] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:21.910 INFO  [atlas-akka.actor.default-dispatcher-7] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:21.916 INFO  [atlas-akka.actor.default-dispatcher-6] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:21.915 INFO  [atlas-akka.actor.default-dispatcher-5] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:21.915 INFO  [atlas-akka.actor.default-dispatcher-3] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:21.915 INFO  [atlas-akka.actor.default-dispatcher-4] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:21.926 INFO  [atlas-akka.actor.default-dispatcher-4] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:21.926 INFO  [atlas-akka.actor.default-dispatcher-7] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:21.926 INFO  [atlas-akka.actor.default-dispatcher-6] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:21.927 INFO  [atlas-akka.actor.default-dispatcher-5] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:21.932 INFO  [atlas-akka.actor.default-dispatcher-5] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:21.932 INFO  [atlas-akka.actor.default-dispatcher-3] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:21.932 INFO  [atlas-akka.actor.default-dispatcher-7] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:21.933 INFO  [atlas-akka.actor.default-dispatcher-4] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:21.933 INFO  [atlas-akka.actor.default-dispatcher-6] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:21.934 INFO  [atlas-akka.actor.default-dispatcher-9] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:21.935 INFO  [atlas-akka.actor.default-dispatcher-8] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:21.936 INFO  [atlas-akka.actor.default-dispatcher-8] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:21.939 INFO  [atlas-akka.actor.default-dispatcher-6] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:21.940 INFO  [atlas-akka.actor.default-dispatcher-3] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:21.940 INFO  [atlas-akka.actor.default-dispatcher-4] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:21.940 INFO  [atlas-akka.actor.default-dispatcher-7] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:21.943 INFO  [atlas-akka.actor.default-dispatcher-7] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:21.943 INFO  [atlas-akka.actor.default-dispatcher-6] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:21.946 INFO  [atlas-akka.actor.default-dispatcher-3] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:21.947 INFO  [atlas-akka.actor.default-dispatcher-4] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:21.951 INFO  [atlas-akka.actor.default-dispatcher-9] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:21.952 INFO  [atlas-akka.actor.default-dispatcher-5] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:21.952 INFO  [atlas-akka.actor.default-dispatcher-9] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:21.952 INFO  [atlas-akka.actor.default-dispatcher-5] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:21.952 INFO  [atlas-akka.actor.default-dispatcher-8] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:21.953 INFO  [atlas-akka.actor.default-dispatcher-8] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:21.957 INFO  [atlas-akka.actor.default-dispatcher-5] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
2017-06-05T19:05:21.957 INFO  [atlas-akka.actor.default-dispatcher-9] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
2017-06-05T19:05:21.957 INFO  [atlas-akka.actor.default-dispatcher-7] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
2017-06-05T19:05:21.957 INFO  [atlas-akka.actor.default-dispatcher-8] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
2017-06-05T19:05:21.957 INFO  [atlas-akka.actor.default-dispatcher-4] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
2017-06-05T19:05:21.959 INFO  [atlas-akka.actor.default-dispatcher-6] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
2017-06-05T19:05:21.964 INFO  [atlas-akka.actor.default-dispatcher-3] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
2017-06-05T19:05:22.999 INFO  [atlas-akka.actor.default-dispatcher-2] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.HealthcheckApi
2017-06-05T19:05:23.000 INFO  [atlas-akka.actor.default-dispatcher-2] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.ConfigApi
2017-06-05T19:05:23.001 INFO  [atlas-akka.actor.default-dispatcher-2] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.akka.StaticPages
2017-06-05T19:05:23.002 INFO  [atlas-akka.actor.default-dispatcher-2] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.TagsApi
2017-06-05T19:05:23.003 INFO  [atlas-akka.actor.default-dispatcher-2] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.RenderApi
2017-06-05T19:05:23.007 INFO  [atlas-akka.actor.default-dispatcher-2] com.netflix.atlas.akka.RequestHandlerActor$$anonfun$loadRoutesFromConfig$1: loading webapi class: com.netflix.atlas.webapi.GraphApi
```
Then, in another session:
```
curl -s 'http://localhost:7101/api/v1/tags'
```
Nothing happens. I confirmed with netstat that port 7101 is not open:
```
$ sudo netstat -tulnp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1322/sshd       
tcp6       0      0 :::22                   :::*                    LISTEN      1322/sshd       
udp        0      0 0.0.0.0:68              0.0.0.0:*  
```

If it's in any way relevant, the system I'm running on is a freshly created AWS EC2 instance, running Ubuntu 16.04, with the latest Oracle Java 8 installed via PPA. I can't find anything wrong in my configuration, but it's entirely possible I don't know what I'm doing with firewall rules.
```
$ java -version
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
```

**Comment by childofsoong** [comment]
When I try to run it locally, I see a line in the log that I don't see in the cloud:
```
2017-06-05T12:49:11.716 INFO  [main] com.netflix.atlas.akka.WebServer: server started on /0:0:0:0:0:0:0:0:7101
[INFO] [06/05/2017 12:49:11.714] [atlas-akka.actor.default-dispatcher-4] [akka://atlas/user/IO-HTTP/listener-0] Bound to /0.0.0.0:7101
```

**Comment by brharrington** [comment]
Thanks for the report. I was able to reproduce using Ubuntu on an EC2 instance. Though strangely it works fine on our test instances.

**Comment by brharrington** [comment]
The fix in #613 clears up the problem for me. Can you verify with `1.5.3-rc.1`?

```
$ curl -LO 'https://github.com/Netflix/atlas/releases/download/v1.5.3-rc.1/atlas-1.5.3-rc.1-standalone.jar'
$ java -jar atlas-1.5.3-rc.1-standalone.jar
```

**Comment by davecromberge** [comment]
I was getting the same on an EC2 instance and the fix above worked for me.  However, when running `publish-test.sh` I received the following:

```
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:48:10       404     0.009
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:48:18       404     0.010
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:48:26       404     0.010
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:48:34       404     0.010
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:48:42       404     0.009
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:48:50       404     0.011
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:48:58       404     0.009
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:49:06       404     0.010
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:49:14       404     0.010
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:49:22       404     0.009
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:49:30       404     0.009
{"type":"error","message":"path not found: /api/v1/publish"}  2017-06-05T22:49:38       404     0.009
```

Not sure if this is related to the changes above or not.

**Comment by brharrington** [comment]
@davecromberge did you run with the `memory.conf`?

https://github.com/Netflix/atlas/wiki/Getting-Started#running-demo-with-memory-storage

By default it has a static data set used for graphing demos and will not configure the publish endpoint.

**Comment by davecromberge** [comment]
@brharrington I copied your command above and did in fact leave out the `memory.conf`.  Looks like it is all working now on my EC2 instance and solves the issue above for me.  Thanks for the quick response!

**Comment by childofsoong** [comment]
@brharrington the fix from `1.5.3-rc.1` does indeed seem to fix my issue. Am I good to close this issue?

**Comment by brharrington** [comment]
Great, thanks for verifying. I'll go ahead and release 1.5.3.
